prSHARK/backends/github.py

```diff
@@ -277,145 +318,161 @@ def parse_pr_list(self, prs):
+            new_pr.target_commit_sha = pr['base']['sha']
+            new_pr.target_commit_id = self._get_commit_id(pr['base']['sha'], new_pr.target_repo_url)
 
-            mongo_pr.merge_commit_id = self._get_commit_id(pr['merge_commit_sha'], mongo_pr.target_repo_url)
+            new_pr.merge_commit_id = self._get_commit_id(pr['merge_commit_sha'], new_pr.target_repo_url)
 
             for lbl in pr['labels']:
-                if lbl['name'] not in mongo_pr.labels:
-                    mongo_pr.labels.append(lbl['name'])
-
-            mongo_pr.save()
-
-            if already_exist:
-                last_update = self._prs.last_updated
-            else:
-                # put the oldest date possible
-                last_update = datetime.datetime(2, 2, 2, 2, 2)
+                if lbl['name'] not in new_pr.labels:
+                    new_pr.labels.append(lbl['name'])
 
             for event in self.fetch_timeline_list(pr['number']):
 
-                if event['event'] == 'committed' and dateutil.parser.parse(
-                        event['committer']['date']).timestamp() > last_update.timestamp():
-
-                    new_commit = True
-                    self.parse_commit(mongo_pr, event)
+                if event['event'] == 'committed':
+                    new_pr.commits.append(event['sha'])
 
-                elif event['event'] == 'reviewed' and dateutil.parser.parse(
-                        event['submitted_at']).timestamp() > last_update.timestamp():
+                elif event['event'] == 'reviewed':
 
                     self.pares_review(mongo_pr, pr, event)
-                elif event['event'] == 'commented' and dateutil.parser.parse(
-                        event['updated_at']).timestamp() > last_update.timestamp():
 
+                elif event['event'] == 'commented':
                     self.parse_comment(event, mongo_pr)
 
-                elif event['event'] == 'assigned' and dateutil.parser.parse(
-                        event['created_at']).timestamp() > last_update.timestamp():
-
-                    mongo_pr.linked_user_ids.append(self._get_person(event['assignee']['url']))
+                elif event['event'] == 'assigned':
 
-                elif event['event'] == 'review_requested' and dateutil.parser.parse(
-                        event['created_at']).timestamp() > last_update.timestamp():
+                    new_pr.linked_user_ids.append(self._get_person(event['assignee']['url']))
 
-                    mongo_pr.requested_reviewer_ids.append(self._get_person(event['requested_reviewer']['url']))
+                elif event['event'] == 'review_requested':
+                    if 'requested_reviewer' in event:
+                        new_pr.requested_reviewer_ids.append(self._get_person(event['requested_reviewer']['url']))
 
-            mongo_pr.save()
+            self.parsed_prs['prs'][self.pr_id] = new_pr
+            self.get_diff(mongo_pr, new_pr, 'pull_request_system_ids')
```

Comment:
Just a remark as the use here is very limited. The way you have written the get_diff method puts the burden to know how it works on the caller of the function due to your use of the exclude_paths. This is usually something you should avoid in larger code bases, instead you can have a method called something like self.save_pr_diff or self.save_pre_diff (save because it does not return something but only saves the marker for differences). Internally those methods could call the generic get_diff method.